### PR TITLE
Common - Add better support for status effects in other components

### PR DESCRIPTION
**Pull request by DartRuffian**
**When merged this pull request will:**
- Add prefixes to status effects, i.e. `"blockDamage"` -> `QGVAR(blockDamage)`
  - BWC for all current status effects
  - Currently status effects added in another component always have `ace_common_` appended to it, see [#10462](https://github.com/acemod/ACE3/pull/10462/files#diff-f398fe1fd48c91ed2eff9cbe29fa19e0c70d11167a043863ae3edf309d87dfdaR139)
- Fix small capitalization mistakes in comments

### IMPORTANT

- If the contribution affects [the documentation](https://github.com/acemod/ACE3/tree/master/docs), please include your changes in this pull request so the documentation will appear on the [website](https://ace3.acemod.org/).
- [Development Guidelines](https://ace3.acemod.org/wiki/development/) are read, understood and applied.
- Title of this PR uses our standard template `Component - Add|Fix|Improve|Change|Make|Remove {changes}`.

### CODE REVIEW

File: addons/common/functions/fnc_statusEffect_get.sqf [hunk 1]
```diff
@@ -26,28 +26,34 @@ if (isNull _object) exitWith {
     [false, []]
 };
 
-[_object, false] call FUNC(statusEffect_resetVariables); //Check for mismatch
+// BWC: Status effect names used to not be prefixed
+if (_effectName in STATUS_EFFECTS_BWC) then {
```
Comment (PabstMirror) [inline addons/common/functions/fnc_statusEffect_get.sqf:30]: this is now case-sensitive, so
```
 [player, "ForceWalk", "myMission", true] call ace_common_fnc_statusEffect_set
```
will break

File: addons/common/functions/fnc_statusEffect_sendEffects.sqf [hunk 2]
```diff
@@ -29,21 +29,20 @@ if (isNull _object) exitWith {};
         //We only do anything if the effect has been defined at some point in the game for this unit
         TRACE_2("checking if event is nil",_x,_effectNumber);
         if (_effectNumber != -1) then {
-            private _eventName = format [QGVAR(%1), _x];
```
Comment (PabstMirror) [inline addons/common/functions/fnc_statusEffect_sendEffects.sqf:32]: Anyone using the old format will now be broken

```
["myThing", true, []] call ace_common_fnc_statusEffect_addType;
["ace_common_myThing", {systemChat str _this}] call CBA_fnc_addEVentHandler;
[player, "myThing", "reason", true] call ace_common_fnc_statusEffect_set;
```
nothing happens

**Review by PabstMirror — CHANGES_REQUESTED** [review]
I think these changes could break things
and just using `QEGVAR(common,field_rations_blockUpdates)` is not that bad

File: addons/common/functions/fnc_statusEffect_sendEffects.sqf (hunk 2 above)
Comment (DartRuffian) [reply addons/common/functions/fnc_statusEffect_sendEffects.sqf:32]: Adding status effects isn't public API, only get/set are

If something was intentionally adding `ace_common_` to a name that feels pretty clear it's only meant to be internally used

**Comment by DartRuffian** [comment]
> I think these changes could break things and just using `QEGVAR(common,field_rations_blockUpdates)` is not that bad

True, but it makes weird naming conventions. It also doesn't make a lot of sense to someone reading through the code until they get to `addType` and see that the event gets `ace_common_` added to the name.

**Comment by PabstMirror** [comment]
Closing in favor or https://github.com/acemod/ACE3/pull/10473

**Pull request closed without merging**